core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -148,6 +150,16 @@ long routedBasedOnClusterVersion() {
         return routedBasedOnClusterVersion;
     }
 
+    /** returns the primary term active at the time the operation was performed on the primary shard */
+    public long primaryTerm() {
+        return primaryTerm;
+    }
+
+    /** marks the primary term in which the operation was performed */
+    public void primaryTerm(long term) {
+        primaryTerm = term;
+    }
```

Comment:
interesting idea - it's currently incremented when relocating a primary though... requires more thought.